multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/util/BasicTokenGenerator.java

```diff
@@ -50,7 +51,9 @@ public OAuth2AccessTokenWithAdditionalInfo generate(String tokenString) {
                                                      StandardCharsets.UTF_8));
         }
         OAuth2AccessTokenWithAdditionalInfo oAuth2AccessTokenWithAdditionalInfo = oauthClient.getToken();
-        storeAccessToken(buildAccessToken(oAuth2AccessTokenWithAdditionalInfo));
+        storeAccessToken(buildAccessToken(oAuth2AccessTokenWithAdditionalInfo),
+                         (String) oAuth2AccessTokenWithAdditionalInfo.getAdditionalInfo()
+                                                                     .get(TokenProperties.USER_ID_KEY));
```

Comment:
can you create a new method in OAuth2AccessTokenWithAdditionalInfo::getUserGuid or something similar